cmd/datamon/cmd/flags.go

```diff
@@ -376,15 +340,14 @@ const (
 	destTNonEmpty
 )
 
-func paramsToDestStore(params flagsT,
-	destT DestT,
+func (flags *flagsT) destStore(destT DestT,
```

Comment:
All in all, my remarks boil down to:
1. the new `cliOptionInput` (without the `cli` prefix...) should bear more responsibilities, and be shed from `cmd`,
2. the `flags` stuff (well indeed, really some Options, since flags is more a CLI-tainted term) should bear a bit less and some more around setting all default values
3 the `CLIConfig` should only be responsible for handling config file and env vars, primarily for use by the CLI, but we should keep in mind that sensitivity to config file and env could be added to lib usage as well (e.g. python bindings). Let's keep that one here for the moment and defer the reshuffling of this CLIconfig.

With such off-loading, the `cmd` package should become an almost trivial package mostly dealing with `cobra`, `core` and `api/params` package or whatever we should call the latter.

Reply:
NOTE: refactoring CLI tests and mocks along the same lines is a story that should also be told, as it remains a non-trival piece of code.

Reply:
yes, going ahead and attaching all methods to the combined config object is the most sensible thing to do, totally agree.

in case there's some question about why it's not that way already, remember that this was _never_ a thoughtful refactor.  it's the result of
* the idea i remarked on in one of your patches to namespace more of the procedures/routines in the cmd package by attaching existing funcs to objects
* inferring some patterns about how to make such a change from the `CLIConfig` implementation
* several quasi-zombie-mode changes wherein i'm saying "consiiiiistencyyy" rather than "braaaiins", and what i'm trying to make consistent is the `CLIConfig` and `flagsT`

Reply:
yes
> the idea i remarked on in one of your patches to namespace more of the procedures/routines in the cmd package by attaching existing funcs to objects
Next, I'll put such things in something like `pkg/api/porcelain`

Reply:
sounds like a plan..

.. i've fallen into a JiT-ing flow myself of late.  i'm okay with anything that makes `cmd/` more maintainable as far as v2 goes.

by v3, 4, or 5, even, perhaps we'll get around to encoding the porcelain in python.  yet this is not a goal in and of itself.  since conversating about this for a while, i've decided that doing so might be a good idea but could also be jumping to conclusions, and relaxed my ask to a more general, "let's try and keep `cmd/` as easy to maintain as possible."